File: Planetary-Formation-main/TerrestrialPlanet.java
import Storage.Positionals.Noise3;
import Storage.ScalarSphere;

public class TerrestrialPlanet {
    private ScalarSphere terrain;

    public void fractalNoise(int depth) {
        Noise3[] layers = new Noise3[depth];

        for (int i = 0; i < depth; i ++) {
            int freq = (int) Math.pow(2, i);
            layers[i] = new Noise3(new int[] {freq, freq, freq});

        }
    }
}

File: Planetary-Formation-main/TerrestrialPlanet.kt
import storage.positionals.Noise3
import storage.positionals.Vector3
import storage.ScalarSphere
import storage.ScalarSphere.SphereApplicationZone
import storage.ScalarSphere.SphereMutation
import storage.positionals.GeoCoord
import kotlin.math.pow

class TerrestrialPlanet internal constructor(resolution: Int, maxRAM: Long) {
    var terrain: ScalarSphere

    init {
        terrain = ScalarSphere(resolution, maxRAM)
    }

    // Scale is the lowest frequency, depth is the highest
    fun initFractalNoise(scale: Int, depth: Int, magnitude: Double) {
        terrain.initFractalNoise(scale, depth, magnitude)
    }

    // Launches one asteroid
    fun launchAsteroid(point: GeoCoord, mass: Double, speed: Double) {

    }
}